js/reference.js

```diff
@@ -73,6 +73,11 @@ var ERMrest = (function(module) {
                 reference._table   = reference._schema.tables.get(reference._tableName);
                 reference._columns = reference._table.columns.all();
 
+                var keys = reference._table.keys.all().sort(function (a, b) {
+                    return a.colset.length() - b.colset.length();
```

Comment:
We might also want to prefer integer/serial keys to text keys for keys of the same length.